vars/updatecli.txt

```diff
@@ -11,6 +11,7 @@
       <li>String updatecliDockerImage: (Optional - Default: "jenkinsciinfra/helmfile:2.3.0") Docker Image of updatecli to be used in the process.</li>
       <li>String cronTriggerExpression: (Optional - Default: "") Enable periodic execution by providing a cron-like expression.</li>
       <li>String containerMemory: (Optional - Default: "512Mi") specify the amount of memory dedicated to the updatecli container.</li>
+      <li>String credentialsId: (Optional - Default: "github-app-updatecli-on-jenkins-infra") specify the githubApp credentials id to use to get an Access Token.</li>
```

Comment:
```suggestion
      <li>String credentialsId: (Optional - Default: "github-app-updatecli-on-jenkins-infra") specify the githubApp credentials id to use to get an access token. The environment variable `UPDATECLI_GITHUB_TOKEN` holds the token value to be used within updatecli.</li>
```

Reply:
I wouldn't add this as this env var is declared only in the code below, while this pipeline parameter doesn't say anything about it.

Reply:
Since the library code is now specifying the name of the env. var (https://github.com/jenkins-infra/pipeline-library/pull/320/files#diff-01eacbe5caf24c385fc8ab8a7684a5580faedb7d729aa4af4d9c3640cf9a4841R60) as part of your change, as a end user, I expect the variable to be documented (however we might have forgotten to document before in parallelDockerUpdatecli). But it's how the end user should expect the token to be used when writing their values.yaml

Reply:
Good point! See https://github.com/jenkins-infra/pipeline-library/pull/320#discussion_r823708232 for a more complete suggestion then :)

Reply:
Oh.. @dduportal can you tell me if we have access to udpatecli/values.yaml data here? If it's the case, I would use the user value instead of hardcoding `UPDATECLI_GITHUB_TOKEN` in the shared pipeline.

Reply:
you have access through the workspace (you can run an `sh` pipeline command and get its output to a variable, or through a groovy method that would read the file from workspace on the agent).

However it's really hard to do: you don't know what keys were used by the end user. It could be directly in the manifest also. That's why I find your current code more elegant as soon as it it documented to end user:
- "If you specify a githubapp credential id, then the variable `UPDATECLI_GITHUB_TOKEN` is set to the token.
- "If you want another variable, then wrap your call to the library `updatecli(...)` inside your own `withEnvironment` and voila"

=> But there is a corner case: if the end user does not have the credential named `github-app-updatecli-on-jenkins-infra` AND does not need to specify another credential. Not probable in our case but that was the reason to delegate credential out of the library itself.

Reply:
> => But there is a corner case: if the end user does not have the credential named github-app-updatecli-on-jenkins-infra AND does not need to specify another credential. 

There is at least another shared pipeline with a default credentialsId value, buildDockerAndPublishImage (which should be checked/updated too in another PR)